modules-js/react-fleet/src/form-elements/inputs/MemorableDateInput.tsx

```diff
@@ -0,0 +1,458 @@
+import React from 'react';
+
+import { css } from 'emotion';
+
+interface Props {
+  initialDate?: string;
+  earliestDate?: string;
+  latestDate?: string;
+  onlyAllowPast?: boolean;
+  onlyAllowFuture?: boolean;
+  legend: React.ReactChild;
+  handleDate: (newDate: Date | null) => void;
+}
+
+interface State {
+  hasFocus: boolean;
+  error: string;
+  date: Date | null;
+  fields: {
+    month: number;
+    day: number;
+    year: number;
+  };
+}
+
+/**
+ * Component to collect a memorable date (such as a birth date) from a user.
+ *
+ * An element must be passed in as the legend, commonly a header <h1-h6>.
+ *
+ * Default behavior is to accept any date. A text string representing a date
+ * can be passed into earliestDate and/or latestDate to limit acceptable input.
+ * Format: MM-DD-YYYY, MM/DD/YYYY, etc. (https://tools.ietf.org/html/rfc2822#section-3.3)
+ *
+ * An initial date can be passed in via the initialDate property.
+ *
+ * onlyAllowPast: If true, will only accept yesterday or earlier for a date.
+ * onlyAllowFuture: If true, will only accept tomorrow or later for a date.
+ *
+ * Whenever the date is updated, this.props.handleDate is called.
+ *
+ * *Please note that all dates are relative to the user’s computer.
+ *
+ * Inspired by https://design-system.service.gov.uk/components/date-input/
+ */
+export default class MemorableDateInput extends React.Component<Props, State> {
+  readonly earliest: Date | null;
+  readonly latest: Date | null;
+  readonly initial: Date | null;
+
+  private timeout: any;
+
+  constructor(props: Props) {
+    super(props);
+
+    this.initial = props.initialDate ? new Date(props.initialDate) : null;
+
+    checkIfPropsValid(props);
+
+    this.earliest = setDateLimit(
+      props.earliestDate || '',
+      !!props.onlyAllowFuture
+    );
+    this.latest = setDateLimit(props.latestDate || '', !!props.onlyAllowPast);
```

Comment:
Do you actually get value out of letting `this.earliest`/`this.latest` to `today` if the `onlyAllow` props are given?

Based on my reading of `isDateValid`, the earliest/latest dates aren't considered if `onlyAllowFuture`/`onlyAllowPast` are set (respectively).

Reply:
If `this.earliest`/`this.latest` are present, they determine the `min` and `max` values for the year number field.